.github/workflows/release.yaml

```
@@ -24,7 +24,7 @@ env:
   PULP_URL: "https://download.konghq.com/kuma-binaries-release/"
   DOCKER_IMAGES: "kumactl,kuma-cp,kuma-dp,kuma-init,kuma-cni"
   RELEASE: ${{ inputs.release || '0.0.0' }}
-  CHECK: ${{ inputs.check }}
+  CHECK: ${{ inputs.check || false }}
```

Is this to convert a string to a boolean?
Maybe
```suggestion
  CHECK: ${{ inputs.check === "true" }}
```
is more clear?

---

inputs.check is a boolean so it's weird as well no? I'm always a little confused with this syntax.

It just seems that CHECK is set to `` when we're running with schedule (because I guess `inputs.check` is unset).
I'm unsure what the type of inputs.check is going to be.

---

Yeah it's terrible. "Booleans" are actually strings :shrug:

---

```
inputs.check === "true"
```

won't work if `inputs.check` is a boolean. I guess we need

```suggestion
  CHECK: ${{ inputs.check == "true" }}
```

---

If it's an empty string or a boolean then we need, otherwise it won't work

```suggestion
  CHECK: ${{ inputs.check.toString() === "true" }}
```

---

Well I actually don't know if it's an empty string or nil :) Let me just try out some stuff in a test repo

---

so you can do 

```suggestion
  CHECK: ${{ inputs?.check?.toString() === "true" }}
```

just to be safe

---

Ok just checked and my original way works:

See this workflow:
https://github.com/lahabana/sandbox/blob/0b417ab439244ee284133783952a4f83a7e4a70f/.github/workflows/action.yaml

Triggered manually with check the if works: https://github.com/lahabana/sandbox/actions/runs/4638055985/jobs/8207516593
Triggered manually without check the if skips:
https://github.com/lahabana/sandbox/actions/runs/4638059419/jobs/8207522343
Triggered via cron the if skips:
https://github.com/lahabana/sandbox/actions/runs/4638181046

I've also tried the original way to make sure I could repro and it failed: https://github.com/lahabana/sandbox/actions/runs/4638630421/jobs/8208615929

The TL;DR I think we can ship it as I first did it.

---

Yeah it's a string. It's just that the `|| false` thing doesn't make this clear. That's why I suggested `=== "true"` which is how you convert a string to a boolean.

---

Looked a little more and when running with schedule the `inputs.check` is actually empty: https://github.com/lahabana/sandbox/commit/d16e0ba3c3b34a3c38a83b9885ad53f924388afe .

I'm going to apply your suggestion @michaelbeaumont.

---

Ok so `===` and `.toString` just don't work. This is not JS it's a subset just for github. Reopening my PR with the original working fix.

---

See: https://github.com/kumahq/kuma/pull/6506